### DISCUSSION TITLE
Feature wishlist

### DISCUSSION THREAD

**Discussion by otseng** [eliranwong/UniqueBible#128] [Ideas]
Is there a feature you wish to see in UBA?  Post it here.

**Comment by eliranwong** [comment]
Below are planned upcoming changes, feel free to comment:

1. add a new tab 'Miscellenous' to 'Master Control' panel with new feature access via Master Control
2. refresh history records when Master Control tab is opened
3. set the search field in focus when Master Control Search tab is opened
4. rename 'Remote Control' to 'mini Control'
5. add an address bar to built-in YouTube downloader window.
- The window will then become practically a simple web browser.
- Two buttons will be added next to the address bar: Download mp3, Download mp4
6. increase the number of verse highlight collections from two to ten
7. add quick access to verse highlight feature via Master Control
8. add ability to customise verse highlight collection name and colours
9. add ability to customise verse number single-click action
10. add ability to customise verse number double-click action
11. add ability to customise default text-to-speech voice via menu
12. add ability to customise javascript for interacting with html content. This may be useful for module creator.

**Reply by eliranwong** [reply]
The above features have been added.  The latest version at the time of writing this post is 21.42.

**Comment by onegladiator4u** [comment]
How about an option to turn off check for update on startup? If someone don't want to update for some time. 
And adding an option to check for update in About/information menu. Forgive me if we already have an option to check for update in other layouts. Haha.

**Reply by otseng** [reply]
Set disableModulesUpdateCheck to True in config options.

**Reply by onegladiator4u** [reply]
I was thinking about UBA app update.

**Reply by otseng** [reply]
This would be a good thing to add.  I was also wondering if people are getting annoyed that the app is getting pretty much being updated every day.

Can you create a feature request for this?

**Reply by onegladiator4u** [reply]
How can I create the request ?

**Reply by otseng** [reply]
Just create an issue and then add the "enhancement" label.

https://github.com/eliranwong/UniqueBible/issues

**Reply by onegladiator4u** [reply]
May I know how to add the label? I couldn't find the option to add it. I think only you guys can add the label. I couldn't find the option on the right side as I see in YouTube videos

**Reply by eliranwong** [reply]
> May I know how to add the label? I couldn't find the option to add it. I think only you guys can add the label. I couldn't find the option on the right side as I see in YouTube videos

you can create an issue, we add a label for you.

**Reply by otseng** [reply]
@eliranwong can you add @onegladiator4u as a reporter?

**Reply by onegladiator4u** [reply]
I created the issue. You can now add label.Enhancement

**Comment by otseng** [comment]
I wish there was a way to quickly see who posted last in the discussions and issues.  There's currently a request to do this.  If anybody else would also like to see this, can you give an amen here?

https://github.com/github/feedback/discussions/2951

**Comment by jdecent** [comment]
Can you make the "Next Chapter" button maintain current view. For example I usually have it in comparison mode. I can advance to next chapter by using "Master Control" window but currently prefer editing the command line input box.

**Reply by jdecent** [reply]
Also, when clicking the verse number the mode is changed from "COMPARE:::" to "BIBLE:::". My expectation was to for it to stay in compare mode and open Smart Indexes

**Reply by jdecent** [reply]
So I have found the option "Enforce Compare / Parallel" which seems to maintain the view as I prefer, but it only works once before switching back to a single bible version.

**Reply by eliranwong** [reply]
@jdecent can you describe more about 'it only works once before switching back to a single bible version'?  I would like to understand more.  thanks.

**Reply by onegladiator4u** [reply]
When we compare Bible versions even if we use "Enforce Compare / Parallel" option, when we click on the verse number more than once it'll go to single Bible version view. It should stay on COMPARE mode. I think that's the issue.

**Reply by eliranwong** [reply]
> When we compare Bible versions even if we use "Enforce Compare / Parallel" option, when we click on the verse number more than once it'll go to single Bible version view. It should stay on COMPARE mode. I think that's the issue.

@onegladiator4u I cannot duplicate the issue you mentioned.  I clicked verse number many times, I still have compare view.  Can you tell me the steps I need to repeat the issue?

**Reply by onegladiator4u** [reply]
[Steps.zip](https://github.com/eliranwong/UniqueBible/files/6183204/Steps.zip)
After the second click to the verse number in COMPARE mode it'll go back to the normal mode as 'BIBLE:::MOV:::Nah 2:10'

**Reply by eliranwong** [reply]
@onegladiator4u thanks for the steps file.  I tried to do the same as follows, but couldn't duplicate the issue, the compare view stays:

1) Click the "Compare / Parallel Enforced" button
2) Open Master Control
3) Select KJV, NET for comparison
4) Click verse numbers many times

Can you help point out if I missed something, in order to duplicate the issue?

**Reply by onegladiator4u** [reply]
Tried with those versions brother. It just not staying in COMPARE Mode.
If you look at the steps you can see that the command line changes to normal 'BIBLE:::MOV:::.......'. just after first click. The next click gets out from COMPARE Mode

**Reply by eliranwong** [reply]
@onegladiator4u can you send the two versions you use in your screenshot to me.

**Reply by onegladiator4u** [reply]
[bibles.zip](https://github.com/eliranwong/UniqueBible/files/6194703/bibles.zip)
@eliranwong Here it is. I tried with  KJV, NET too. As you suggested.

**Reply by jdecent** [reply]
It stays in compare mode only once, then second verse number selection returns to KJV. Also does not open smart indexes in study window as it does when only viewing KJV. I noticed that after clicking the first number(15) the compare box changes to "BIBLE:::KJV:::Rom 3.15" which does not reflect the compare view I see in the bible window
[UniqueBible.app [version 22.95] - UniqueBible.app 2021-03-24 01-14-53.zip](https://github.com/eliranwong/UniqueBible/files/6194831/UniqueBible.app.version.22.95.-.UniqueBible.app.2021-03-24.01-14-53.zip)

**Reply by eliranwong** [reply]
@jdecent @onegladiator4u It is very interesting it always stays on compare view on my side.  Here is my screen recording.  It doesn't matter command field is changing back to BIBLE:::, as it will be ignored, as you can see my screen recording.

I am still puzzled why the difference, before I can find a fix.  as you can see in screen recording, I couldn't duplicate it.  but I can see the issue in @jdecent screen recording.   I need to think more why the difference
[Screen recording 2021-03-24 10.14.37 AM.zip](https://github.com/eliranwong/UniqueBible/files/6196210/Screen.recording.2021-03-24.10.14.37.AM.zip)
.

**Reply by onegladiator4u** [reply]
Ok. Take your time. Here's my debug info and config file details if you want it.

`==============================================================
Debugging Information
==============================================================
2021-03-24 16:36:48.766543

# Platform
Python=['3.9.1 (tags/v3.9.1:1e5d33e, Dec  7 2020, 17:08:21) [MSC v.1927 64 bit (AMD64)]']
Could not get uname
platform=Windows-10-10.0.19041-SP0
platform=nt
platform=10

# Locale
timezone=India Standard Time+0530
defaultlocale=('en_IN', 'cp1252')
Could not get locale

# Version
UBA Version: 23.09

# Config
config.theme=dark
config.userLanguage=Malayalam
config.userLanguageInterface=False
config.menuLayout=focus
config.theme=dark
config.menuShortcuts=micron
config.marvelData=marvelData
config.qtMaterial=True
config.qtMaterialTheme=dark_lightgreen.xml
config.useFastVerseParsing=True
config.menuShortcuts=micron
config.startupMacro=
config.enableMacros=False
config.useFastVerseParsing=True
config.disableModulesUpdateCheck=False

# Marvel data
Bibles: 24
Books: 26
Data: 13
Commentaries: 15
Lexicons: 15`

[zip file deleted here, in order to protect your personal api key from the public.]

**Reply by otseng** [reply]
I can duplicate it.  I'll fix it here:

https://github.com/eliranwong/UniqueBible/issues/303

**Reply by eliranwong** [reply]
I watch @jdecent screen recording once more.  I finally figured when it does not work and why the difference between my computer setting and yours @jdecent @onegladiator4u 

I found that you have Sync Study Window Bible enabled.  When it is enabled, config.enforceCompareParallel does not work.  I will try to fix along this finding.

The latest PR has an issue, that's why I haven't updated that to end-users yet.

meanwhile, you can use enforceCompareParallel, by disabling sync Study Window bible.

I will to look into a more satisfactory solution.

**Reply by eliranwong** [reply]
Fixed in this commit: https://github.com/eliranwong/UniqueBible/commit/665b56474ff078d2ec9d36eb5d61319daa50a654

**Reply by eliranwong** [reply]
@jdecent @onegladiator4u you may update to version 23.10 to get the fix, thanks for reporting the issue.

**Reply by onegladiator4u** [reply]
It's working . Thanks for the quick fix.

**Comment by onegladiator4u** [comment]
Hi guys,
I got mp3 files for my language Bible . I also have the information on the starting and ending of the timing for the verse of the Bible chapter.
Is there a chance that you could make files that can be used along mp3 files that import timings like these to play each verse than full chapter. Otherwise we have to cut each chapter mp3 to verses mp3.
i have html that has info like this

var timings = [
    { label: "1", start: 3.5, end: 21.05 },
    { label: "2", start: 21.3, end: 32.35 },
    { label: "3", start: 32.6, end: 36.83 },
    { label: "4", start: 37.08, end: 47.33 },
    { label: "5", start: 47.58, end: 57.55 },
    { label: "6", start: 57.8, end: 68.59 },
    { label: "7", start: 68.84, end: 79.59 },
    { label: "8", start: 79.84, end: 85.91 },
    { label: "9", start: 86.16, end: 96.63 },
    { label: "10", start: 96.88, end: 101.51 },
    { label: "11", start: 101.76, end: 110.99 },
    { label: "12", start: 111.24, end: 122.35 },
    { label: "13", start: 122.6, end: 132.91 },
    { label: "14", start: 133.16, end: 158.95 },
    { label: "15", start: 159.2, end: 168.27 },
    { label: "16", start: 168.52, end: 179.83 },
    { label: "17", start: 180.08, end: 192.03 },
    { label: "18", start: 192.28, end: 198.59 },
    { label: "19", start: 198.84, end: 215.79 },
    { label: "20", start: 216.04, end: 225.59 },
    { label: "21", start: 225.84, end: 234.31 },
    { label: "22", start: 234.56, end: 240.79 },
    { label: "23", start: 241.04, end: 249.19 },
    { label: "24", start: 249.44, end: 258.67 },
    { label: "25", start: 258.92, end: 269.91 },
    { label: "26", start: 270.16, end: 289.19 },
    { label: "27", start: 289.44, end: 298.75 },
    { label: "28", start: 299, end: 314.35 },
    { label: "29", start: 314.6, end: 324.03 },
    { label: "30", start: 324.28, end: 334.79 },
    { label: "31", start: 335.04, end: 350.03 },
    { label: "32", start: 350.28, end: 362.99 },
    { label: "33", start: 363.24, end: 382.35 },
    { label: "34", start: 382.6, end: 402.11 },
    { label: "35", start: 402.36, end: 411.87 },
    { label: "36", start: 412.12, end: 426.03 },
    { label: "37", start: 426.28, end: 442.53 },
    { label: "38", start: 442.78, end: 448.71 },
    { label: "39", start: 448.96, end: 457.05 },
    { label: "40", start: 457.3, end: 469.83 },
    { label: "41", start: 470.08, end: 481.31 },
    { label: "42", start: 481.56, end: 488.59 },
    { label: "43", start: 488.84, end: 503.79 },
    { label: "44", start: 504.04, end: 520.31 }
];
If this is very a big job don't need to look into this.

**Reply by otseng** [reply]
I'll look into this.

https://github.com/eliranwong/UniqueBible/issues/1034

**Comment by onegladiator4u** [comment]
Hey guys,
Since material layout now have easy access to chapter on toolbar like this
![image](https://user-images.githubusercontent.com/48617390/171992485-b9e476ea-faa1-4559-9add-4a59059c2cfa.png)

Does this layout need chapter numbers on top and bottom of main window like this? May be you could add option to hide it.
![image](https://user-images.githubusercontent.com/48617390/171992579-4d558b99-4b1e-46c0-8da9-a4df111d6652.png)

What do you guys think?

**Reply by otseng** [reply]
Created issue: https://github.com/eliranwong/UniqueBible/issues/1057

**Comment by eliranwong** [comment]
@onegladiator4u Hi bro, do you use Windows?  can you send me a screenshot of UBA system tray on Windows?  Thank you.

**Reply by onegladiator4u** [reply]
@eliranwong Hi bro,
Yes, i'm using windows.
Here it is 
![image](https://user-images.githubusercontent.com/48617390/172003959-12d0542c-9139-4910-998f-4b4e3effa9a8.png)
Just so you know it won't open UBA if we click on it. And have to close UBA through system tray right clicking. It would be nice if there's an option to close it when we click on window close.

**Reply by eliranwong** [reply]
@onegladiator4u can you take a screenshot when menu is open like the macOS or Linux screenshot in this wiki page: https://github.com/eliranwong/UniqueBible/wiki/UBA-System-Tray ?  Many thanks.

**Reply by onegladiator4u** [reply]
> @onegladiator4u can you take a screenshot when menu is open like the macOS or Linux screenshot in this wiki page: https://github.com/eliranwong/UniqueBible/wiki/UBA-System-Tray ? Many thanks.

Oh sorry. i thought you just want to look how it looks.

![image](https://user-images.githubusercontent.com/48617390/172005409-57200c26-714b-47f1-92ec-67e1d7edc9f0.png)

**Reply by eliranwong** [reply]
Great!  Thank you.

**Reply by eliranwong** [reply]
just curious, do you have two UBA copies?  Is the red icon also UBA?

**Reply by onegladiator4u** [reply]
> just curious, do you have two UBA copies? Is the red icon also UBA?

Haha no, it's Bitdefender Antivirus :-)

**Comment by pvt02** [comment]
Hello. I just found out about this program, and I am trying to figure out how to install it. Here is my request: please make an appimage of this program to make it easier to use. Appimages just work, and are the simplest to use on Linux; even common peons can use them right away. I am using an old bblx file of yours on a wine version of E-sword, which led me here. Thanks.

**Comment by 1eam** [comment]
Hi there, 
Are there developments to get this as executable?
One that could run out of the box on a usb drive for example.

That way it would be portable and userfriendly. Happy coding! Phillipians 4:13

**Reply by otseng** [reply]
I had tried that.  But unfortunately, the executable gets blocked by virus detection programs.  So, it had to be abandoned.

If you want some other ways to use it without installing anything, you can go to:
https://bible.gospelchurch.uk/index.html
https://transliteralbible.com/index.html

A mobile friendly one is at:
https://uniquebibleapp.com

**Reply by 1eam** [reply]
Give it another go! try again.
Also not everyone uses a virus-detection programme.
I for instance dont and havent used one in a decade. Besides many anti-virusses provide an option to trust a piece of software.

However that asside, I'm sure there are options to get your programme installed as regular .exe or mac installation which may require a fundamental change and therefore even a separate rebuild.

Perhaps another framework
Don't give up! Happy coding :)

**Reply by pvt02** [reply]
Thanks for the feedback from other users. I am a recovering Windows user, and recommend Devuan Linux, with the Cinnamon desktop, and a Windows 7 theme.  This OS is more secure (Systemd is being cut out -from the NSA/RedHat-, just saying ...) than even most other Linux varieties. It is also not too hard to learn to use as a former Windows user. If you try it do not give up! Also try learning to use virtual machines for even more security. ANYway, thanks again to the developers of UniqueBible.

**Comment by pvt02** [comment]
Hello again. I have been looking for the morphology/grammar codes KEY for marvel.bible

I also notice that https://hb.openscriptures.org/structure/OshbVerse/index.html?b=Gen&c=1&v=1
has different morphology/grammar codes. For example for the word-pair b/ruhshith in Genesis 1.1 : there the codes are 
HR/Ncfsa and the KEY is located at
https://hb.openscriptures.org/parsing/HebrewMorphologyCodes.html
which then gives the meaning as Hebrew pReposition/Noun common-noun feminine singular absolute-state, if I understand properly.

But for the same word-pair at marvel.bible there is: prep subs.f.sg.a
I can guess some of these (I for example do not know what "subs" might mean), but for reading/studying/learning Hebrew scripture we need a KEY. Can you please provide a link to that key? There are people who can benefit from your work that have less study than I do, and they would not make heads of tails of these codes.

I also notice that the first site uses "b" for the preposition meaning "in" etc, but marvel.bible uses some sort of extended Strong's H9003. Where can we see this extended Strongs concordance?

**Reply by pvt02** [reply]
update: So on the issue of morphology/grammar codes, yesterday on marvel.bible when I clicked the codes nothing happened. Today something happened, a new window popped up on the right, with a brief explanation. However that key/explanation still left questions, such as how "subs." means a noun, or what "unknown" refers to.

**Comment by pvt02** [comment]
Hello again. Donations are cool to help your project; but I have never got this really working on my Linux setup. It would be nice to be able to donate toward making an appimage version somehow, so we (me and others) can just download it, mark it as executable allowed, and run it.

**Reply by eliranwong** [reply]
> Hello again. Donations are cool to help your project; but I have never got this really working on my Linux setup. It would be nice to be able to donate toward making an appimage version somehow, so we (me and others) can just download it, mark it as executable allowed, and run it.

Hi, I think installation on Linux is the most straightforward.  May I ask which specific difficulties you encountered?

**Comment by pvt02** [comment]
I installed the program successfully, but after closing the window, it would not restart. This problem persisted despite weeks of troubleshooting (I am not good at this stuff). Though I switched from Windows to Devuan Linux and am willing to invest considerable time making software work, even Louis Rossman acknowledges the time-consuming nature of Linux configuration. I understand many computer experts look down on users who view computers as mere tools, preferring Windows or Apple interfaces, but I disagree with this perspective. My BASH knowledge exists only out of necessity as a Windows refugee on Linux. While open source software promotes freedom, alienating users who need practical functionality undermines community growth.

Usability Concerns and Suggestions

Play On Linux proved more effective for installing E-Sword than using command line instructions for your program. Though your software functioned one time after blindly following complex instructions over a year ago, an appimage format would simply work immediately, and always, on all flavors of Linux. If financial resources permit, I would fund conversion of your software to an appimage format. Many potential users could benefit from Hebrew and Greek parallel Bible tools without requiring system administration expertise-similar to how construction workers use tools without becoming repair technicians. The appimage approach (download, double-click, and begin studying) eliminates unnecessary technical barriers, as demonstrated by FreeCad, and many other Linux programs, which I recently began learning immediately after downloading its appimage. If you ever make this a paid appimage; I will buy it though it has some flaws.

Hebrew Grammar Inaccuracies

Your Hebrew grammar notes contain inaccuracies requiring correction. For instance, אֵת is labeled merely as an "object marker" when it specifically functions as a "definite direct object marker." Additionally, שָׁמַיִם receives the simple gloss "heavens" without explaining its status as a plurale tantum requiring context-dependent translation as either singular or plural despite its plural form. The notation "subs.m.pl.a H8064" provides insufficient linguistic information for students relying on your notes to learn Hebrew accurately.
